### PR TITLE
Using no_std + alloc

### PR DESCRIPTION
**Pull request by halzy**
Removed references to `std` and replaced them with references to `core` and `alloc`.

Tests still use `std`.

### CODE REVIEW

File: CHANGELOG.md
Comment (lambda-fairy) [inline CHANGELOG.md:61]: These changes look unrelated. Can you please revert everything except the one changelog entry?

File: maud/src/lib.rs
Comment (lambda-fairy) [inline maud/src/lib.rs:15]: Can we remove this? This module doesn't define any unit tests afaik.

File: maud/src/lib.rs
Comment (lambda-fairy) [inline maud/src/lib.rs:182]: Does this work?

I'm pretty sure `core` doesn't have an `io` module.

You might want to disable `no_std` when these web framework features are enabled, since they require `std` anyway.

File: maud_macros/src/lib.rs
Comment (lambda-fairy) [inline maud_macros/src/lib.rs:39]:
```
error[E0433]: failed to resolve: could not find `alloc` in the list of imported crates
  --> maud/tests/misc.rs:66:20
   |
66 |     let result_a = html! { (r) };
   |                    ^^^^^^^^^^^^^ not found in `alloc::string`
   |
```

You can probably fix this by adding `extern crate alloc;` above the existing `extern crate maud;` line.

**Review by lambda-fairy — CHANGES_REQUESTED** [review]
Thanks for the PR!

Happy to merge this once the issues above are resolved.

BTW, if you'd like a quicker turnaround next time, I recommend running `cargo test --workspace --all-features` locally before sending the PR. That should catch most of these issues.

File: maud/src/lib.rs
Comment (halzy) [reply maud/src/lib.rs]: I didn't find a way to optionally disable `no_std`, only a way to optionally enable it.

**Comment by halzy** [comment]
> Thanks for the PR!
> 
> Happy to merge this once the issues above are resolved.
> 
> BTW, if you'd like a quicker turnaround next time, I recommend running `cargo test --workspace --all-features` locally before sending the PR. That should catch most of these issues.

Ah, this would have been helpful. I'll open a second PR for updating the `CONTRIBUTING.md` file with this information.

File: maud/src/lib.rs
Comment (halzy) [reply maud/src/lib.rs]: This was updated to re-enable std for rocket and iron support.

File: CHANGELOG.md
Comment (halzy) [reply CHANGELOG.md]: done.

**Comment by halzy** [comment]
@lambda-fairy CI should pass. Let me know if there are other changes you'd like.

**Review by lambda-fairy — APPROVED** [review]
LGTM (with a couple minor fixes I made myself)

Thanks again @halzy!

**Comment by lambda-fairy** [comment]
Oh, I notice GitHub requires maintainer approval before it runs CI. I've changed it to run automatically instead.

**Pull request merged**
